File: MixItApp/MixItApp/Common/ProjectManager.m
//
//  ProjectManager.m
//  Aacmultitrack
//
//  Created by LMAN on 1/31/16.
//  Copyright © 2016 idragon. All rights reserved.
//

#import "ProjectManager.h"
#import "Common.h"
#import "AMTTrackManager.h"

@implementation ProjectManager

+ (NSString*)newProject {
    
    NSString * newProjectPath = [Common createNewProjectDocumentPath];
    return newProjectPath;
}

+ (NSMutableArray*)getProjectList {
    
    NSFileManager * fileManager = [NSFileManager defaultManager];
    NSArray * fileList = [fileManager contentsOfDirectoryAtPath:[Common getProjectsDocumentPath] error:nil];

    NSMutableArray * projectArray = [[NSMutableArray alloc] init];
    for (NSString * fileName in fileList) {
        
        BOOL isDirectory;
        NSString * projectDirPath = [[Common getProjectsDocumentPath] stringByAppendingPathComponent:fileName];
        BOOL isFileExist = [fileManager fileExistsAtPath:projectDirPath isDirectory:&isDirectory];
        
        if (isDirectory && isFileExist) {
            [projectArray addObject:fileName];
        }
    }
    
    return projectArray;
}

+ (NSMutableArray*)openProject:(NSString*)projectName {
    
    NSString * projectPath = [Common getProjectPath:projectName];
    NSFileManager * fileManager = [NSFileManager defaultManager];
    NSArray * fileList = [fileManager contentsOfDirectoryAtPath:projectPath error:nil];
    
    NSMutableArray * trackArray = [[NSMutableArray alloc] init];
    for (NSString * fileName in fileList) {
        
        BOOL isDirectory;
        NSString * audioDirectoryPath = [projectPath stringByAppendingPathComponent:fileName];
        BOOL isFileExist = [fileManager fileExistsAtPath:audioDirectoryPath isDirectory:&isDirectory];
        
        if (isDirectory && isFileExist) {
            
            AMTTrack * track = [[AMTTrack alloc] init];
            NSArray * trackFileList = [fileManager contentsOfDirectoryAtPath:audioDirectoryPath error:nil];
            if (trackFileList.count > 0) {
                
                NSString * trackFileName = trackFileList[0];
                track.trackName = trackFileName;
                track.audioURL = [NSURL fileURLWithPath:[audioDirectoryPath stringByAppendingPathComponent:trackFileName]];
                track.trackVolume = 1.0f;
                [trackArray addObject:track];
            }
        }
    }
    
    return trackArray;
}

+ (void)deleteProject:(NSString*)projectName {
    
    NSString * projectPath = [Common getProjectPath:projectName];
    [[NSFileManager defaultManager] removeItemAtPath:projectPath error:nil];
}

@end

File: MixItApp/MixItApp/Common/ProjectManager.swift
//
//  ProjectManager.h

//  Project manager class for multitrack audio file.
//
import Foundation
class ProjectManager: NSObject {
    class func newProject() -> NSString {
        let newProjectPath = Common.createNewProjectDocumentPath()
        return newProjectPath
    }
    
    class func openProject(projectName: String) -> [AnyObject] {
        let projectPath: String = Common.getProjectPath(projectName)
        let fileManager: NSFileManager = NSFileManager.defaultManager()
        var trackArray: [AnyObject] = [AnyObject]()
        do {
            let fileList = try fileManager.contentsOfDirectoryAtPath(projectPath)
            
            for fileName: String in fileList {
                let audioDirectoryPath: String = (projectPath as NSString).stringByAppendingPathComponent(fileName)
                let isFileExist: Bool = fileManager.fileExistsAtPath(audioDirectoryPath)
                if isFileExist {
                    let track = AMTTrack()
                    do {
                        let trackFileList = try fileManager.contentsOfDirectoryAtPath(audioDirectoryPath)
                        if trackFileList.count > 0 {
                            let trackFileName: String = trackFileList[0]
                            track.trackName = trackFileName
                            track.audioURL = NSURL.fileURLWithPath((audioDirectoryPath as NSString).stringByAppendingPathComponent(trackFileName))
                            track.trackVolume = 1.0
                            trackArray.append(track)
                        }
                    }
                    catch{
                        
                    }
                    
                }
            }
        }
        catch{
            
        }
        
        return trackArray
    }
    
    class func deleteProject(projectName: String) {
        let projectPath: String = Common.getProjectPath(projectName)
        do {
            try NSFileManager.defaultManager().removeItemAtPath(projectPath)
        }
        catch{
            
        }
        
    }
    
    class func getProjectList() -> [AnyObject] {
        let fileManager: NSFileManager = NSFileManager.defaultManager()
        do {
            let fileList = try fileManager.contentsOfDirectoryAtPath(Common.getProjectsDocumentPath())
            var projectArray: [AnyObject] = [AnyObject]()
            for fileName: String in fileList {
                let projectDirPath: String = (Common.getProjectsDocumentPath() as NSString).stringByAppendingPathComponent(fileName)
                let isFileExist: Bool = fileManager.fileExistsAtPath(projectDirPath)
                if isFileExist {
                    projectArray.append(fileName)
                }
            }
            return projectArray
        }
        catch{
            return []
        }
        
    }
}